### PR TITLE
adding missing RIDs

### LINKED ISSUE #829: RID catalog is out of date
It seems [the RID catalog](https://docs.microsoft.com/en-us/dotnet/articles/core/rid-catalog#using-rids) is not up to date with 1.0 RTM. But I'm not sure which RIDs should be listed here, so I'm creating this issue to discuss that.

[Here is a list of all the RIDs](https://gist.github.com/svick/eb084a68e50547f36b1022a50f3f2982), based on [Microsoft.NETCore.Platforms/runtime.json](https://github.com/dotnet/corefx/blob/release/1.0.0/pkg/Microsoft.NETCore.Platforms/runtime.json). 

Ignoring RIDs that can't actually be installed on a machine (e.g. `any`, `win` or `win7`), RIDs that are not listed fall within several categories. For each category, it should be decided whether to list it in the catalog or not.

* Windows 8+ ARM: I think those should be included, assuming they are supported.
* Windows 8.1: Windows 7, 8 and 10 are currently listed, 8.1 isn't. It should probably be included, since someone asked about it in a comment.
* Windows AOT: No idea.
* Oracle Linux (`ol`): Should be included.
* CentOS: `centos.7.1-x64` is included in the catalog, but does not seem to actually be a valid RID, it should be removed.
* Debian: `debian.8.2-x64` is in the same situation as CentOS, should be removed.
* Ubuntu: 15.10 and 16.04 should be added.
* Fedora: Should be added.
* OpenSUSE: Should be added.

### PR DESCRIPTION
**Pull request by mairaw**
Fixes #829, #923 

Pending questions:
- Windows AOT RIDs: these are for .NET Native. Should I add them to the docs?

/cc @richlander @blackdwarf

### CODE REVIEW

File: docs/core/rid-catalog.md [hunk 1]
```diff
@@ -107,19 +114,33 @@ wish to check if new ones are added, please check back here.
...
+    * `ol.7.2-x64`
 * Currently supported Ubuntu derivatives 
     * `linuxmint.17-x64`
     * `linuxmint.17.1-x64`
     * `linuxmint.17.2-x64`
     * `linuxmint.17.3-x64`
+    * `linuxmint.18-x64`
 
-## OS X RIDs
+## Mac RIDs
```
Comment (svick) [inline docs/core/rid-catalog.md]: Should this say "macOS" per https://github.com/dotnet/core-docs/pull/612#discussion_r68465747?

Comment (mairaw) [inline docs/core/rid-catalog.md]: what do you think @richlander? Should we keep OS X as you suggested back in that earlier comment? I was following how we're calling the OSes on this page for this change: https://www.microsoft.com/net/core

**Comment by mairaw** [comment]
Updated the doc based on comments from @piotrpMSFT and @ericstj. Thanks guys!

There's still a question on whether we want to list the AOT RIDs at this point or not, but this is ready for review if we don't.

/cc @blackdwarf

**Comment by blackdwarf** [comment]
@ericstj are AOT RIDs targetable?

**Comment by ericstj** [comment]
What do you mean by targetable?  No RIDs are "targetable" today because NuGet doesn't allow for different compile time assemblies based on RID (what we typically refer to when we say targetable).  RIDs control what runtime assemblies are used.  We do indeed use the aot RIDs for UWP applications when that application is using the .NETNative toolchain.

**Comment by mairaw** [comment]
@ericstj so I should add them here to this doc too? Can people use that in their json files already in their .NET Core projects?

**Comment by ericstj** [comment]
Not for .NET Core, only UWP.

**Comment by blackdwarf** [comment]
@ericstj I meant are they "usable" for a, for instance, self-contained deployment, to which you gave an answer above. OK, so since these are .NET Core docs, I would suggest not putting them in and keeping with those we can use in .NET Core projects.

**Comment by mairaw** [comment]
Yes, I agree with this plan @blackdwarf. And what about the changes? Do they look OK to you?

**Comment by blackdwarf** [comment]
:shipit:

**Pull request merged**